src/vector.js

```diff
@@ -53,6 +154,17 @@ Vector.cross  = function (v1, v2)
     ];
 };
 
+// Dot product of two vectors
+Vector.dot = function (v1, v2)
+{
+    var n = 0.0;
+    var lim = Math.min(v1.length, v2.length);
```

Comment:
It actually gets better, because testing to see if you have a NaN value is hard. In JS
```javascript
NaN === NaN; // false
NaN == NaN; // false
```
The correct way to detect if you have a NaN value is to use Object.is, now. I think this is new.
```javascript
Object.is(NaN, NaN); // true
```